drop-transfer/src/ws/client/socket.rs

```diff
@@ -0,0 +1,56 @@
+use std::{
+    io,
+    time::{Duration, Instant},
+};
+
+use futures_util::{SinkExt, StreamExt};
+use tokio::net::TcpStream;
+use tokio_tungstenite::{tungstenite::Message, WebSocketStream};
+
+pub type WsStream = WebSocketStream<TcpStream>;
+
+pub struct WebSocket {
+    stream: WsStream,
+    recv_last: Instant,
+    recv_timeout: Duration,
+    send_timeout: Duration,
+}
+
+impl WebSocket {
+    pub fn new(stream: WsStream, recv_timeout: Duration, send_timeout: Duration) -> Self {
+        Self {
+            stream,
+            recv_last: Instant::now(),
```

Comment:
To be pedantic I think `Option<Instant>` would be slightly more correct

Reply:
Oh, yeah. True. You gave me a thought that I'm not setting this field in the receive() method. My mistake

Reply:
Fixed